src/nvim/msgpack_rpc/channel.c

```diff
@@ -776,15 +587,16 @@ static bool is_rpc_response(msgpack_object *obj)
 static bool is_valid_rpc_response(msgpack_object *obj, Channel *channel)
 {
   uint64_t response_id = obj->via.array.ptr[1].via.u64;
+  ChannelCallFrame *frame = kv_last(channel->rpc.call_stack);
+
   // Must be equal to the frame at the stack's bottom
-  return kv_size(channel->call_stack) && response_id
-    == kv_A(channel->call_stack, kv_size(channel->call_stack) - 1)->request_id;
+  return kv_size(channel->rpc.call_stack)
```

Comment:
If the stack size can be zero then `kv_last()` should not be used for `frame` (negative array index, UB?).